rfd/0107-abandon-active-access-request.md

```diff
@@ -0,0 +1,103 @@
+---
+authors: Jeff Pihach (jeff.pihach@goteleport.com)
+state: draft
+---
+
+# RFD 0107 - Abandon Active Access Requests
+
+## Required Approvers
+
+- Engineering: @zmb3
+
+## What
+
+Allow users who are using elevated permissions via an access request to abandon
+them ahead of their expiry once they are no longer of use.
+
+## Why
+
+By reducing the amount of time that a user has elevated permissions we also
+reduce the time and surface area to which an attack could occur in the event
+the users device becomes compromised.
+
+## Success Criteria
+
+- A user is able to abandon an active access request ahead of their expiry time
+  via the CLI or UI.
+- Creating a lock to prevent further use of the existing certificate.
```

Comment:
I dont think we need to do this - we can keep locks separate. If you want to lock an access request, there's already a mechanism to do so.

This feature is simply for saying "I don't need this anymore, mark it complete"

Reply:
Alright, in discussing it with others it was unclear that "abandoning" an access request early wouldn't also lock out any active sessions. But maybe that's a terminology thing with the name "abandon" and if we used something like "finished" as you mentioned below would alleviate this.

Reply:
I think "abandoning" an access request without applying a lock is pretty much equivalent to saying "I promise I deleted all copies of the certs and I won't use them anymore, trust me". There is no real guarantee of anything. Without a lock, or equivalent, I get the feeling that this feature adds some confusion without adding any real security benefit